pyani_plus/db_orm.py

```diff
@@ -88,6 +96,61 @@ def __repr__(self) -> str:
         )
 
 
+class Configuration(Base):
+    """Describes the configuration of a pyANI-plus run.
+
+    Each run is a set of all-vs-all comparisons between a set of FASTA
+    files (genome table rows), for a specific ANI algorithm or method,
+    recorded as a set of comparison table rows.
+
+    Each Run entry represents a specific instance of an analysis run which
+    records details like the exact command line, and when it was run. It will
+    point to a single Configuration entry (this class) which records the tool
+    name, version, and parameters like k-mer size (some parameters are tool
+    specific).
+
+    Comparison entries are cached for unique combinations of the query and
+    subject genomes (via the hash of their file contents, not via filenames),
+    AND the run configuration.
+    """
+
+    __tablename__ = "configuration"
+
+    __table_args__ = (
+        UniqueConstraint(
+            "method",
+            "program",
+            "version",
+            "fragsize",
+            "maxmatch",
+            "kmersize",
+            "minmatch",
+        ),
+    )
+
+    configuration_id = Column(Integer, primary_key=True)
+
+    # This was part of the Run table in pyANI v0.2
+    method = Column(String)
+    # These were all part of the Comparison table in pyANI v0.2, which had
+    # them as part of the uniqueness constraint.
+    program = Column(String)
+    version = Column(String)
+    fragsize = Column(Integer)  # in fastANI this is fragLength
+    maxmatch = Column(Boolean)  # in fastANi this is Null
```

Comment:
Change this to a dict/JSON of parameters in a single field? Likely also to affect kmersize/minmatch/fragsize columns?

We need to consider whether we're going to query the db on each parameter - this would mean one column per parameter, with db schema/version updates managed by Alembic. More complex, but also more useful for comparisons between runs.

Reply:
We could still have an "additional parameters" column for the other parameters for a run.